flict/flictlib/project/reader.py

```diff
@@ -0,0 +1,229 @@
+        if Project.dependencies_license(package).strip() != "":
+            return f" ( {Project.package_license(package)} ) AND ( {Project.dependencies_license(package)} ) "
+        else:
+            return Project.package_license(package)
+
+
+class ProjectReader:
+    """Interface for classes that reads project from some kind of format"""
+    def read_project(self):
+        return
+
+    def _add_files_read(self, project_file):
+        self.files_read.append(project_file)
+
+    def _already_read(self, project_file):
+        return project_file in self.files_read
+
+    def _flatten_packages(self, packages):
+        package_list = []
+
+        for _k, v in packages.items():
+            _package = v
+            tmp_dict = {}
+
+            for dep in _package['dependencies']:
+                tmp_dict.update(self._flatten_package_tree(dep))
+
+            dep_list = []
+            for dep in tmp_dict.values():
+                dep_list.append(dep)
+
+            package = {
+                'name': _package['name'],
+                'version': _package['version'],
+                'license': _package['license'],
+                'description': _package['description'],
+                'dependencies': dep_list
+            }
+
+            package_list.append(package)
+        return(package_list)
+
+
+class ProjectReaderFactory:
+
+    @staticmethod
+    def get_projectreader(project_file=None, project_dirs = None, project_format=None):
+        if not project_dirs:
+            project_dirs = ["."]
+        logging.debug(f'get_projectreader({project_file}, {project_dirs}, {project_format})')
+        #suffix = os.path.splitext(project_file)[-1].lower()
+        if project_format is None:
+            if "spdx" in project_file:
+                return SPDXJsonProjectReader(project_dirs)
+            if "flict" in project_file:
+                return FlictProjectReader(project_dirs)
+        elif project_format == "flict":
+            return FlictProjectReader(project_dirs)
+        elif project_format == "spdx":
+            return SPDXJsonProjectReader(project_dirs)
+
+
+class FlictProjectReader(ProjectReader):
+    """Class for reading flict project files"""
+
+    def __init__(self, project_dirs):
+        pass
+
+    def read_project(self, project_file):
+        try:
+            with open(project_file, 'r') as f:
+                project_data = json.load(f)
+                project = project_data['project']
+
+                return {
+                    "project_name": project['name'],
+                    "packages": [
+                        {
+                            'name': project['name'],
+                            'version': project['version'],
+                            'license': project['license'],
+                            'description': '',
+                            'dependencies': project['dependencies']
+                        }
+                    ]
+                }
```

Comment:
This part could be moved out of the try catch block.

Reply:
will keep this to create a flict exception if the project file is faulty (e.g. missing key)